language/move-lang/src/naming/translate.rs

```diff
@@ -0,0 +1,766 @@
+                        None => {
+                            assert!(context.has_errors());
+                            NE::UnresolvedError
+                        }
+                        Some(b) => NE::Builtin(sp(n.loc, b), nes),
+                    },
+                    Some((m, f)) => NE::ModuleCall(m, f, ty_args, nes),
+                },
+                sp!(_, EE::ExpDotted(ed)) => match *ed {
+                    dexp!(_) => panic!("ICE stand alone expdotted"),
+                    dot!(inner, f) => match *inner {
+                        dexp!(sp!(_, EE::ModuleIdent(m))) => {
+                            NE::ModuleCall(m, FunctionName(f), ty_args, nes)
+                        }
+                        edot => {
+                            let ndot = dotted(context, edot);
+                            NE::MethodCall(ndot, FunctionName(f), ty_args, nes)
+                        }
+                    },
+                },
+                sp!(loc, _) => {
+                    context.error(vec![
+                        (eloc, "Invalid function call"),
+                        (loc, "Expected: a name, a dotted Module access, or dotted access of an expression")
+                    ]);
+                    NE::UnresolvedError
+                }
+            }
+        }
+        EE::ModuleIdent(n) => {
+            context.error(vec![
+                    (eloc, format!("Unexpected module identifier: '{}'", n)),
+                    (n.loc(), "Modules can only be used in expressions as the left hand side of a '.' for a function call".to_string())
+                ]);
+            NE::UnresolvedError
+        }
+
+        EE::UnresolvedError => {
+            assert!(context.has_errors());
+            NE::UnresolvedError
+        }
+    };
+    sp(eloc, ne_)
+}
+
+fn dotted(context: &mut Context, edot: E::ExpDotted) -> N::ExpDotted {
+    let sp!(loc, edot_) = edot;
+    let nedot_ = match edot_ {
+        E::ExpDotted_::Exp(e) => N::ExpDotted_::Exp(exp(context, e)),
+        E::ExpDotted_::Dot(d, f) => N::ExpDotted_::Dot(Box::new(dotted(context, *d)), Field(f)),
+    };
+    sp(loc, nedot_)
+}
+
+fn bind(context: &mut Context, b: E::Bind) -> Option<N::Bind> {
+    use E::Bind_ as EB;
+    use N::Bind_ as NB;
+    let sp!(loc, b_) = b;
+    let nb_ = match b_ {
+        EB::Var(v) => {
+            if v.starts_with_underscore() {
+                NB::Ignore
+            } else {
+                NB::Var(v)
+            }
+        }
+        EB::Unpack(tn, tys_opt, efields) => {
+            let (m, sn) = context.resolve_struct_name(tn)?;
+            let nfields = UniqueMap::maybe_from_opt_iter(
+                efields
+                    .into_iter()
+                    .map(|(k, (idx, inner))| Some((k, (idx, bind(context, inner)?)))),
+            )?;
+            NB::Unpack(
+                m,
+                sn,
+                tys_opt.map(|tys| base_types(context, tys)),
+                nfields.unwrap(),
```

Comment:
`.ok()?` instead of `unwrap`?

Reply:
Its an ICE if its not Err. added a comment